java/sage/FileDownloader.java

```diff
@@ -318,7 +320,7 @@ else if (remoteUIXfer)
         outStream.flush();
         str = Sage.readLineBytes(inStream);
         remoteSize = Long.parseLong(str.substring(0, str.indexOf(' ')));
-        fileOut = new FastRandomFile(myDestFile, "rw", Sage.I18N_CHARSET);
+        fileOut = new LocalSageFile(myDestFile, false);
```

Comment:
FastRandomFile does buffered writes, so you should probably wrap this in the buffering class for equivalence.